core/src/main/java/org/radargun/SlaveBase.java

```diff
@@ -73,6 +74,17 @@ protected void scenarioLoop() throws IOException {
             } else if (!stage.shouldExecute()) {
                log.info("Stage '" + stage.getName() + "' should not be executed");
                response = new DistStageAck(state);
+               if (stage instanceof AbstractDistStage) {
```

Comment:
You're right of course; I thought that `getExecutingSlaves` is called from `shouldExecute`. Yes, checking that in shouldExecute is the solution I'd prefer.